pkg/operator/starter.go

```diff
@@ -209,3 +218,48 @@ func extractOperatorStatus(obj *unstructured.Unstructured, fieldManager string)
 	}
 	return &ret.Status.OperatorStatusApplyConfiguration, nil
 }
+
+func withCustomEndPoint(infraLister v1.InfrastructureLister) dc.DeploymentHookFunc {
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		infra, err := infraLister.Get(infrastructureName)
+		if err != nil {
+			return err
+		}
+		if infra.Status.PlatformStatus == nil || infra.Status.PlatformStatus.PowerVS == nil {
+			return nil
+		}
+		serviceEndPoints := infra.Status.PlatformStatus.PowerVS.ServiceEndpoints
+		if len(serviceEndPoints) == 0 {
+			return nil
+		}
+		var containerEnvVars []corev1.EnvVar
+		for _, serviceEndPoint := range serviceEndPoints {
+			containerEnvVars = append(containerEnvVars, corev1.EnvVar{
```

Comment:
We are interested only in 3 endpoints, The service endpoint may contain several other, So I think we should set only these 3.

Reply:
Sure.. I'll have the changes made.